jaxlib/jax.bzl

```diff
@@ -186,41 +180,109 @@ def if_building_jaxlib(
         "//conditions:default": [],
     })
 
-def _get_test_deps(deps, backend_independent):
+def _get_jaxlib_test_deps(deps, backend_independent):
     gpu_build_deps = [
         "//jaxlib/cuda:gpu_only_test_deps",
         "//jaxlib/rocm:gpu_only_test_deps",
         "//jax_plugins:gpu_plugin_only_test_deps",
     ]
+    common_test_deps = py_deps([
+        "absl/testing",
+        "numpy",
+        "ml_dtypes",
+        "scipy",
+        "opt_einsum",
+        "hypothesis",
+        "cloudpickle",
+        "flatbuffers",
+    ])
 
     gpu_py_imports = [
-        "//:jax_py_import",
         "//jaxlib/tools:jaxlib_py_import",
         "//jaxlib/tools:jax_cuda_plugin_py_import",
         "//jaxlib/tools:jax_cuda_pjrt_py_import",
-    ]
+    ] + common_test_deps
     cpu_py_imports = [
-        "//:jax_py_import",
         "//jaxlib/tools:jaxlib_py_import",
-    ]
+    ] + common_test_deps
+    jaxlib_pypi_wheel_deps = [
+        "@pypi//jaxlib",
+    ] + common_test_deps
 
     if backend_independent:
         jaxlib_build_deps = deps
-        gpu_pypi_wheel_deps = _CPU_PYPI_WHEEL_DEPS
+        gpu_pypi_wheel_deps = jaxlib_pypi_wheel_deps
         gpu_py_import_deps = cpu_py_imports
     else:
         jaxlib_build_deps = gpu_build_deps + deps
-        gpu_pypi_wheel_deps = _GPU_PYPI_WHEEL_DEPS
+        gpu_pypi_wheel_deps = jaxlib_pypi_wheel_deps + [
+            "@pypi//jax_cuda12_plugin",
+            "@pypi//jax_cuda12_pjrt",
+        ]
         gpu_py_import_deps = gpu_py_imports
 
     return select({
         "//jax:enable_jaxlib_build": jaxlib_build_deps,
-        "//jax_plugins/cuda:disable_jaxlib_for_cpu_build": _CPU_PYPI_WHEEL_DEPS,
+        "//jax_plugins/cuda:disable_jaxlib_for_cpu_build": jaxlib_pypi_wheel_deps,
         "//jax_plugins/cuda:disable_jaxlib_for_cuda12_build": gpu_pypi_wheel_deps,
         "//jax_plugins/cuda:enable_py_import_for_cpu_build": cpu_py_imports,
         "//jax_plugins/cuda:enable_py_import_for_cuda12_build": gpu_py_import_deps,
     })
 
+def _get_jax_test_deps():
+    jax_build_deps = [
```

Comment:
I don't understand why you need this list. If you need build targets, then I would only include the core targets plus whatever is in the deps list explicitly. Indeed, I would expect that you do *not* pull most of these into most tests; they would need explicit deps.

Reply:
Added filtering by `@pypi//`.